packages/task/src/browser/task-service.ts

```diff
@@ -238,6 +247,24 @@ export class TaskService implements TaskConfigurationClient {
         }
     }
 
+    /**
+     * Run the selected text lines in active terminal.
+     */
+    async runSelectedText(): Promise<void> {
+        if (!this.editorManager.currentEditor) { return; }
+        const startLine = this.editorManager.currentEditor.editor.selection.start.line;
+        const endLine = this.editorManager.currentEditor.editor.selection.end.line;
+        const selectedRange: Range = Range.create(startLine, 0, endLine + 1, 0);
```

Comment:
Why whole line? in VS Code it runs only selection:
```ts
const selectedText = this.editorManager.currentEditor.editor.document.getText(this.editorManager.currentEditor.editor.selection);
```

Reply:
My mistake. I thought VS Code behaved differently. This has been fixed.
After some testing, there are two general cases that must be covered:

1) User selects no text - VS code sends entire line that the cursor is on to the terminal
2) User selects text - VS code only sends the portion of text that is selected

I think the last commit now handles both cases.